### DISCUSSION TITLE
Allow 'getStaticProps' and 'getServerSideProps' on the same page

### DISCUSSION THREAD

**Discussion by abidk** [vercel/next.js#11919] [Help]
At the moment it's not possible to use getStaticProps and getServerSideProps on the same page. It would appear you have to use getStaticProps and then do client side requests to handle the dynamic data.

I think there is a valid use case, where you want to build a page with initial static data and then allow SSR on page requests where the data is dynamic.

**Comment by timneutkens** [comment]
Sounds like you're asking for https://github.com/zeit/next.js/discussions/11552 instead.

**Reply by mshez** [reply]
@timneutkens  the use case is following.
I have dynamic page page/example/[slug].tsx . For SEO of page i have to call API request which will get me meta title and description (same for all dynamic slugs). I want to do that on build (getStaticProps). While rest of page as is dynamic i want to use getServerSideProps.

Solution can be that i call the API on serverSide everytime. But that will be extra api call on every request while i need it to be available on build time.  Hope you understand the scenario

**Comment by ajjack50n** [comment]
I have a use case where I am trying to build a form based on a Sequel model and then I would like to populate the form on every page request.  This would be a nice way of keeping a html form in sync with the backend data model.
